### DISCUSSION TITLE
Welcome to KernelMemoryService Discussions!

### DISCUSSION THREAD

**Discussion by marcominerva** [marcominerva/KernelMemoryService#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by nurkmez2** [comment]
Hi Marco,

I'm reaching out regarding the Kernel Memory project. ([https://github.com/microsoft/kernel-memory](https://github.com/microsoft/kernel-memory))

I've been using it in production for about two months now — especially as a standalone service via the Docker image. However, I've noticed there haven't been any updates lately (for about 3 months now), and I'm growing concerned about whether the project is actively maintained.

As an example, I’ve seen what happened with other Microsoft projects like Chat Copilot, which appear to be going stale. ([https://github.com/microsoft/chat-copilot](https://github.com/microsoft/chat-copilot))

Could you please share whether Kernel Memory is still being developed or maintained long-term?

If not, would you recommend an alternative from Microsoft (or others) that has similar capabilities?

Thank you for your time and all your contributions!

Best regards

**Reply by marcominerva** [reply]
You can find some information about the project here: https://github.com/microsoft/kernel-memory/discussions/532#discussioncomment-13814424.

In this moment, I have deciced to create a custom solution like https://github.com/marcominerva/SqlDatabaseVectorSearch, but this doesn't offer the same capabilities of Kernel Memory (it is an ad-hoc solution).

**Reply by nurkmez2** [reply]
Hi Marco,

Thank you for the clarification and for sharing the link.

Kernel Memory has been a great product — I’ve really enjoyed working with it in production, and it would be a shame to see it abandoned. I truly hope it won’t be left to “fade away,” as it fills a valuable niche.

I’ll take a look at your custom solution as well, though I agree that KM’s capabilities are quite unique.

Thanks again for all your efforts and contributions!

Best regards,

**Comment by nurkmez2** [comment]
Hi Marco,

I wanted to ask for your honest opinion about Kernel Memory v2.
Do you see it as a truly promising and more solid evolution compared to v1,
or do you think there’s still a risk that it might face a similar fate in the medium term?

I’d really appreciate your perspective, especially regarding real-world adoption and the overall direction of the project.

Thanks a lot, and great work!

**Reply by marcominerva** [reply]
I think that at this stage it’s difficult to understand where the project will go. For the moment, though, I tend to be cautious, also because the README clearly states: “This is an active research project. It is evolving rapidly and may change without notice.” So for now, in my opinion, it’s best to wait.

**Reply by nurkmez2** [reply]
Hi Marco,

Thanks a lot for your honest perspective — really appreciate you taking the time to share it.
That makes perfect sense, and I’ll follow your advice and keep a cautious approach for now.

Happy Holidays / Merry Christmas and wishing you a great New Year! 🎄✨

Best regards,
